pkg/networkservice/common/discover/match_selector.go

```diff
@@ -73,22 +73,28 @@ func matchEndpoint(clockTime clock.Clock, nsLabels map[string]string, ns *regist
 	return validNetworkServiceEndpoints
 }
 
-// ProcessLabels generates matches based on destination label selectors that specify templating.
-func ProcessLabels(str string, vars interface{}) string {
+// processLabels generates matches based on destination label selectors that specify templating.
+func processLabels(str string, vars interface{}) string {
 	tmpl, err := template.New("tmpl").Parse(str)
 
 	if err != nil {
-		panic(err)
+		return str
 	}
-	return process(tmpl, vars)
+
+	rv, err := process(tmpl, vars)
+	if err != nil {
+		return str
+	}
+
+	return rv
 }
 
-func process(t *template.Template, vars interface{}) string {
+func process(t *template.Template, vars interface{}) (string, error) {
 	var tmplBytes bytes.Buffer
 
 	err := t.Execute(&tmplBytes, vars)
 	if err != nil {
-		panic(err)
+		return "", err
```

Comment:
Good catch on not panicing here.  Could we wrap the error so we get some context?